### PR TITLE
Apply some simplifications and deduplications to ExactTime sync policy

### PR DESCRIPTION
**Pull request by cwecht**

### CODE REVIEW

File: include/message_filters/sync_policies/exact_time.hpp [hunk 1]
```diff
@@ -154,40 +154,42 @@ struct ExactTime : public PolicyBase<Ms...>
...
-        tuples_.erase(old);
-      } else {
-        // the map is sorted by time, so we can ignore anything after this if this one's time is ok
-        break;
-      }
-    }
+    const auto arrived_before_last_signal = [this](const auto & e)
+      {return e.first <= last_signal_time_;};
+    const auto first_arrived_after =
+      std::partition_point(tuples_.begin(), tuples_.end(), arrived_before_last_signal);
```
Comment (ahcorde) [inline include/message_filters/sync_policies/exact_time.hpp:171]: include `<algorithm>`

File: include/message_filters/sync_policies/exact_time.hpp [hunk 2]
```diff
@@ -154,40 +154,42 @@ struct ExactTime : public PolicyBase<Ms...>
...
+    drop_range(tuples_.begin(), first_arrived_after);
   }
 
 private:
+  using M_TimeToTuple = std::map<rclcpp::Time, Tuple>;
+  using map_iterator = typename M_TimeToTuple::iterator;
+
+  void call_drop_signal(Tuple & t)
+  {
+    std::apply([this](auto &&... args) {this->drop_signal_.call(args ...);}, t);
```
Comment (ahcorde) [inline include/message_filters/sync_policies/exact_time.hpp:181]: include `tuple`

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21604/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15892/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1108/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22322/)

File: include/message_filters/sync_policies/exact_time.hpp (hunk 1 above)
Comment (cwecht) [reply include/message_filters/sync_policies/exact_time.hpp]: Not needed anymore: it turns out, that `std::map::upper_bound` is a way better solution in this case.

File: include/message_filters/sync_policies/exact_time.hpp (hunk 2 above)
Comment (cwecht) [reply include/message_filters/sync_policies/exact_time.hpp:177]: Is already included (line 36).

**Comment by MichaelOrlov** [comment]
@mjcarroll @sloretz Friendly ping here.
We agreed to dismiss a stale review and merge this PR since all comments were addressed and CI was green.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**